File: bindings/uniffi/lib/java/io/gorules/zen_engine/JsonBuffer.java
package io.gorules.zen_engine;

import org.jetbrains.annotations.NotNull;

import java.nio.charset.StandardCharsets;

public record JsonBuffer(byte[] value) {
    public JsonBuffer(String value) {
        this(value.getBytes(StandardCharsets.UTF_8));
    }

    @NotNull
    @Override
    public String toString() {
        return new String(value, StandardCharsets.UTF_8);
    }
}

File: bindings/uniffi/lib/kotlin/io/gorules/zen_engine/kotlin/JsonBuffer.kt
package io.gorules.zen_engine.kotlin

@JvmInline
value class JsonBuffer(val value: ByteArray) {
    constructor(json: String) : this(json.toByteArray(Charsets.UTF_8))

    override fun toString(): String = value.toString(Charsets.UTF_8)
    fun toByteArray(): ByteArray = value
}